lib/vector-core/proto/event.proto

```diff
@@ -95,7 +107,7 @@ message Metric {
   }
   string namespace = 11;
   uint32 interval_ms = 18;
-  Value metadata = 19;
+  Metadata metadata = 19;
```

Comment:
Note to self- this is a breaking change and needs to be documented/ process followed

Reply:
💬 I suggest that we deprecate the existing field and add a new field instead to maintain backwards/forwards compatibility. Changing the type breaks backward compatibility, making Vector more difficult to operate.

Once fully deprecated, we will mark `19` as a reserved index.

https://earthly.dev/blog/backward-and-forward-compatibility/

Reply:
this has been addressed

Reply:
Good call on forward maintaining compatibility here 👍